RGPosadas/LitterRally
# TASK-85: Modal Component

closes #301

**NOTE: Changes in app.tsx are TEMPORARY, and will be reverted before the pr is merged. See commit messages for details.**

This pr aims to create a generic modal component for use throughout the app. It also creates a FriendModal which will be useful for upcoming tickets.

How to test:
- There is some temporary code I added in app.tsx that you can play around with to try the modals
```tsx
      {userData && true && (
        <FriendModal
          open={open}
          setOpen={setOpen}
          paperClasses="asdf-asdf"
          user={userData.user as User}
          button1Text="hello"
          button1OnClick={() => {
            console.log('asdf');
          }}
          button2Text="hello"
          button2OnClick={() => {
            console.log('asdf');
          }}
        ></FriendModal>
      )}
      {false && (
        <MullModal open={open} setOpen={setOpen}>
          Hello There!
        </MullModal>
      )}
```
- When checking out to my branch for the first time, once you log in a refresh the page, you should see a modal pop up with your username and picture, and two buttons. You can play around with the props of the FriendModal to change the content.
![image](https://user-images.githubusercontent.com/23544999/112741744-3e4b1700-8f56-11eb-9bf2-579bb9ba4605.png)

- If you want to see how the generic MullMoral component looks, you can set `{userData && true && (` to `{userData && false && (` to hide the FriendModal, and set `{false && (` to `{true && (` to show the MullModal. You can insert any jsx content in the modal, so go wild and try out stuff

![image](https://user-images.githubusercontent.com/23544999/112741757-4f942380-8f56-11eb-813d-09ae47ac388e.png)
